pumpkin-protocol/src/client/play/c_commands.rs

```diff
@@ -0,0 +1,353 @@
+                let mut n = 2;
+                if override_suggestion_tpye.is_some() {
+                    n |= Self::FLAG_HAS_SUGGESTION_TYPE
+                }
+                if is_executable {
+                    n |= Self::FLAG_IS_EXECUTABLE
+                }
+                n
+            }
+        };
+        bytebuf.put_u8(flags);
+
+        // child count + children
+        bytebuf.put_list(&self.children, |bytebuf, child| bytebuf.put_var_int(child));
+
+        // redirect node
+        if flags & Self::FLAG_HAS_REDIRECT != 0 {
+            bytebuf.put_var_int(&1.into());
+        }
+
+        // name
+        match self.node_type {
+            ProtoNodeType::Argument { name, .. } | ProtoNodeType::Literal { name, .. } => {
+                bytebuf.put_string(name)
+            }
+            ProtoNodeType::Root => {}
+        }
+
+        // parser id + properties
+        if let ProtoNodeType::Argument {
+            name: _,
+            is_executable: _,
+            parser,
+            override_suggestion_type: _,
+        } = &self.node_type
+        {
+            parser.write_to_buffer(bytebuf)
+        }
+
+        if flags & Self::FLAG_HAS_SUGGESTION_TYPE != 0 {
+            match &self.node_type {
+                ProtoNodeType::Argument { name: _, is_executable: _, parser: _, override_suggestion_type } => {
+                    // suggestion type
+                    let suggestion_type = &override_suggestion_type.expect("ProtoNode::FLAG_HAS_SUGGESTION_TYPE should only be set if override_suggestion_type is not None.");
+                    bytebuf.put_string(suggestion_type.identifier());
+                },
+                _ => unimplemented!("ProtoNode::FLAG_HAS_SUGGESTION_TYPE is only implemented for ProtoNodeType::Argument"),
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum ProtoCmdArgParser<'a> {
+    Bool,
+    Float { min: Option<f32>, max: Option<f32> },
+    Double { min: Option<f64>, max: Option<f64> },
+    Integer { min: Option<i32>, max: Option<i32> },
+    Long { min: Option<i64>, max: Option<i64> },
+    String(StringProtoArgBehavior),
+    Entity { flags: u8 },
+    GameProfile,
+    BlockPos,
+    ColumnPos,
+    Vec3,
+    Vec2,
+    BlockState,
+    BlockPredicate,
+    ItemStack,
+    ItemPredicate,
+    Color,
+    Component,
+    Style,
+    Message,
+    Nbt,
+    NbtTag,
+    NbtPath,
+    Objective,
+    ObjectiveCriteria,
+    Operation,
+    Particle,
+    Angle,
+    Rotation,
+    ScoreboardSlot,
+    ScoreHolder { flags: u8 },
+    Swizzle,
+    Team,
+    ItemSlot,
+    ItemSlots,
+    ResourceLocation,
+    Function,
+    EntityAnchor,
+    IntRange,
+    FloatRange,
+    Dimension,
+    Gamemode,
+    Time { min: i32 },
+    ResourceOrTag { identifier: &'a str },
+    ResourceOrTagKey { identifier: &'a str },
+    Resource { identifier: &'a str },
+    ResourceKey { identifier: &'a str },
+    TemplateMirror,
+    TemplateRotation,
+    Heightmap,
```

Comment:
loot_table, loot_predicate and loot_modifier are missing. They are below Heightmap

Reply:
![image](https://github.com/user-attachments/assets/d469026f-1b00-4d87-854b-6e64f38c6c1c)

I even looked at this, no idea how I could miss that. But they're added now